### PR TITLE
simplify version check and add more tests

### PR DESCRIPTION
**Pull request by ramaraochavali**
Simplifies version check and adds more tests

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/model/context.go [hunk 1]
```diff
@@ -641,11 +648,6 @@ func ParseServiceNodeWithMetadata(s string, metadata *NodeMetadata) (*Proxy, err
 
 // ParseIstioVersion parses a version string and returns IstioVersion struct
 func ParseIstioVersion(ver string) *IstioVersion {
-	if strings.HasPrefix(ver, "master-") {
-		// This proxy is from a master branch build. Assume latest version
-		return MaxIstioVersion
-	}
-
 	// strip the release- prefix if any and extract the version string
 	ver = istioVersionRegexp.FindString(strings.TrimPrefix(ver, "release-"))
```
Comment (ramaraochavali) [inline pilot/pkg/model/context.go]: @howardjohn does Istio Version for release start with "release-" ?

Comment (howardjohn) [reply pilot/pkg/model/context.go]: we used to, same with master-, until 1.4

Comment (ramaraochavali) [reply pilot/pkg/model/context.go]: ok. I think we should leave it like that for now and may be remove later. 1.5 pilot still should process "release-" envoys.

**Pull request merged**